### DISCUSSION TITLE
Custom CDN and server

### DISCUSSION THREAD

**Discussion by prasanna1211** [vercel/next.js#16344] [Help]
So to explore the flexibility of Next.js, I would like to understand the feasibility of hosting a Next JS app but without Vercel. 

This would include

1. Running the next JS app in Docker in an EC2 instance.
2. AWS cloudfront CDN that caches the HTML pages for SSG. (or any other CDN would do). Instead of vercel doing this out of the box, I would like to host it with some other CDN.

Is this kind of set up possible at the first place ?

Also what would we lose if we host Next JS app on our own.

**Comment by ats1999** [comment]
I am open for discussions regarding this problem, would love to connect with you.
